sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OtlpConfigUtil.java

```diff
@@ -19,6 +19,18 @@
   static final String DATA_TYPE_TRACES = "traces";
   static final String DATA_TYPE_METRICS = "metrics";
 
+  static String getOtlpProtocol(String dataType, ConfigProperties config) {
+    String protocol =
+        config.getString(String.format("otel.experimental.exporter.otlp.%s.protocol", dataType));
+    if (protocol == null) {
+      protocol = config.getString("otel.experimental.exporter.otlp.protocol");
+    }
+    if (protocol == null) {
+      protocol = "grpc";
+    }
+    return protocol;
```

Comment:
Totally NIT: `return (protocol == null ? grpc : protocol)` - as an idiom for "default value"